packages/gatsby/src/schema/data-tree-utils.js

```diff
@@ -14,24 +23,36 @@ const flatten = require(`flat`)
  *
  * @param {*Nodes} args
  */
-export const extractFieldExamples = (nodes: any[]) => {
+const extractFieldExamples = (nodes: any[]) => {
   // $FlowFixMe
-  return _.mergeWith({}, ...nodes, (obj, next) => {
-    if (!_.isArray(obj || next)) return
-    let array = [].concat(obj, next).filter(v => v != null)
+  return _.mergeWith({}, ...nodes, (obj, next, key, o, s, stack) => {
+    if (obj === INVALID_VALUE) return obj
+
+    // TODO: if you want to support infering Union types this should be handled
+    // differently. Maybe merge all like types into examples for each type?
+    // e.g. union: [1, { foo: true }, ['brown']] -> Union Int|Object|List
+    if (!isSameType(obj, next)) return INVALID_VALUE
```

Comment:
Yeah let's. The vast majority of Gatsby users will never see it think about these issues. They'll just add source and transformer plugins and get going. So it's important that for the few creating plugins that we insist they get things right so little mistakes made there don't reverberate across everyone downstream.